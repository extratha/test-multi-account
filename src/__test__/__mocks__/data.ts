export const data = {
  test: 'test'
}