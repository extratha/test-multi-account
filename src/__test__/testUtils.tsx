import { RenderOptions, RenderResult, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { SWRConfig } from 'swr';
import { theme } from '@/config/config-mui/theme';
import { ThemeProvider as MUIThemeProvider } from '@mui/material/styles';
import { NextIntlClientProvider } from 'next-intl';
import commonTh from '../../public/locales/th/common.json';
import commonEn from '../../public/locales/th/common.json';
type RenderWithProvider = (
  elm: React.ReactElement,
  renderOptions?: ProviderOptions,
) => RenderResult;

export type ProviderOptions = {
  formOptions?: Record<string, unknown>;
} & RenderOptions;

const messages = {
  ...commonTh,
  ...commonEn,
};

const renderWithProviders: RenderWithProvider = (
  component,
  renderOptions?,
  locale = 'th',
) => {
  const wrapper = ({ children }: { children: React.ReactNode }) => (
    <NextIntlClientProvider locale={locale} messages={messages}>
      <MUIThemeProvider theme={theme}>
        <SWRConfig value={{ provider: () => new Map() }}>{children}</SWRConfig>
      </MUIThemeProvider>
    </NextIntlClientProvider>
  );

  return render(component, {
    wrapper,
    ...renderOptions,
  });
};

export { renderWithProviders as render, userEvent };
export * from '@testing-library/react';
