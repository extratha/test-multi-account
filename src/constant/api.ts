
export const API = {
  PATH: {
    login: 'auth/login',
    setNewPassword: 'auth/reset-password',
    changePassword: 'auth/change-password',
    forgotPassword: 'auth/forgot-password',
    exampleDataList: 'lab/examples',
  },
  HTTP_METHODS: {
    GET: 'GET',
    POST: 'POST',
    PUT: 'PUT',
    PATCH: 'PATCH',
    DELETE: 'DELETE',
  },
  ERROR: {
    UNAUTHENTICATION: {
      MESSAGE: 'unauthenticated',
      CODE: 401,
    },
    login: {
      INVALID_CREDENTIAL: {
        MESSAGE: 'invalid email or password',
        CODE: 400,
      },
    }
  },
};
