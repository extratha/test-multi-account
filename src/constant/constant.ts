export const GENDER = {
  MALE: "Male",
  FEMALE: "Female",
};

export const CONFIG_FIELD_TYPES = {
  STRING: "String",
  NUMBER: "Number",
  DROPDOWN: "Dropdown",
};
