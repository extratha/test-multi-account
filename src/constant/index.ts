export const ENV = {
  BASE_API_URL: process.env.NEXT_PUBLIC_BASE_API_URL || "MOCK_NEXT_PUBLIC_BASE_API_URL",
  FIREBASE_API_KEY: process.env.NEXT_PUBLIC_FIREBASE_API_KEY || "MOCK_NEXT_PUBLIC_FIREBASE_API_KEY",
  FIREBASE_AUTH_DOMAIN: process.env.NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN || "MOCK_NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN",
  FIREBASE_PROJECT_ID: process.env.NEXT_PUBLIC_FIREBASE_PROJECT_ID || "MOCK_NEXT_PUBLIC_FIREBASE_PROJECT_ID",
  FIREBASE_STORAGE_BUCKET:
    process.env.NEXT_PUBLIC_FIREBASE_STORAGE_BUCKET || "MOCK_NEXT_PUBLIC_FIREBASE_STORAGE_BUCKET",
  FIREBASE_MESSAGING_SENDER_ID:
    process.env.NEXT_PUBLIC_FIREBASE_MESSAGING_SENDER_ID || "MOCK_NEXT_PUBLIC_FIREBASE_MESSAGING_SENDER_ID",
  FIREBASE_APP_ID: process.env.NEXT_PUBLIC_FIREBASE_APP_ID || "MOCK_NEXT_PUBLIC_FIREBASE_APP_ID",
  FIREBASE_MEASUREMENT_ID:
    process.env.NEXT_PUBLIC_FIREBASE_MEASUREMENT_ID || "MOCK_NEXT_PUBLIC_FIREBASE_MEASUREMENT_ID",
};

export const SESSION = {
  ACCESS_TOKEN: "access-token",
  REFRESH_TOKEN: "refresh-token",
  RESET_PASSWORD_TOKEN: "reset-password-token",
  X_ROBOT_TOKEN: "x-robot-token",
};

export const MENU_CONDITION = {
  LOGOUT: "logout",
};

export const NAVIGATION = {
  NOT_FOUND: "/not-found",
  LOGIN: "/login",
  HOME: "/home",
  FORGET_PASSWORD: "/forget-password",
  SET_NEW_PASSWORD: "/set-new-password",
  CONSENT_TERMS_CONDITIONS: "/consents/terms-conditions",
  CONSENT_PRIVACY_POLICIES: "/consents/privacy-policies",
  AI_INTERPRET_RESULT: "/ai-interpret/interpret",
  AI_INTERPRET_TRY_EXAMPLE_DATA: "/ai-interpret/try-example-data",
  AI_INTERPRET_TRY_INPUT_DATA: "/ai-interpret/try-input-data",
  SYMPTOM_CHECKER: "/symptom-checker",
  SETTINGS_TERMS_AND_CONDITIONS: "/settings/terms-and-conditions",
  SETTINGS_PRIVACY_POLICY: "/settings/privacy-policy",
  PUBLIC_TERMS_AND_CONDITIONS: "/public/consents/terms-conditions",
  PUBLIC_PRIVACY_POLICY: "/public/consents/privacy-policies",
};

export const GENDER = {
  MALE: "Male",
  FEMALE: "Female",
};

export const CONFIG_FIELD_TYPES = {
  STRING: "String",
  NUMBER: "Number",
  DROPDOWN: "Dropdown",
};

export const GROUP_NAME = {
  GENERAL_CHECK_UP: "generalCheckUp",
  HEMATOLOGY_BG: "hematologyBG",
  HEMATOLOGY_CBC: "hematologyCBC",
  LIPID_PROFILE: "lipidProfile",
  BLOOD_CHEMISTRY: "bloodChemistry",
  RENAL_FUNCTION_TEST: "renalFunctionTest",
  LIVER_FUNCTION_TEST: "liverFunctionTest",
  URINE_ANALYSIS: "urineAnalysis",
  STOOL_EXAMINATION: "stoolExamination",
  PAP_SMEAR: "papSmear",
  TUMOR_MARKER: "tumorMarker",
};

export const GENERAL_CHECK_UP = {
  AGE: "age",
  GENDER: "gender",
  BODY_WEIGHT: "body_weight",
  BODY_HEIGHT: "body_height",
  WAIST_LINE: "waist_line",
  BLOOD_PRESSURE_SYSTOLIC: "blood_pressure_systolic",
  BLOOD_PRESSURE_DIASTOLIC: "blood_pressure_diastolic",
  PULSE: "pulse",
  RESPIRATORY_RATE: "respiratory_rate",
  TEMPERATURE: "temperature",
};

export const HEMATOLOGY_BLOOD = {
  BLOOD_GROUP: "blood_group",
  BLOOD_GROUP_RH: "blood_group_rh",
};

export const HEMATOLOGY_BLOOD_GROUP = {
  HEMATOLOGY_BLOOD: "bloodGroup",
};

export const HEMATOLOGY_CBC = {
  HB: "hb_value",
  HCT: "hct_value",
  RBC_MORPHOLOGY: "rbc_morphology_value",
  MCV: "mcv_value",
  MCH: "mch_value",
  MCHC: "mchc_value",
  RDW: "rdw_value",
  WBC: "wbc_value",
  NEUTROPHIL: "neutrophil_value",
  LYMPHOCYTE: "lymphocyte_value",
  MONOCYTE: "monocyte_value",
  EOSINOPHIL: "eosinophil_value",
  BASOPHIL: "basophil_value",
  PLATELET_COUNT: "platelet_count_value",
};

export const HEMATOLOGY_CBC_GROUP = {
  HEMOGLOBIN: "hemoglobin",
  WHITE_BLOOD: "whiteBlood",
  RED_BLOOD: "redBlood",
};

export const INTERPRET_STATUS = {
  PENDING: "pending",
  SUCCESS: "success",
  FAILED: "failed",
};

export const ASR_SERVICE = {
  ORDER_ASR: "order-asr",
  SOAP_ASR: "soap-asr",
};

export const ASR_SERVICE_KEY: Record<string, string> = {
  [ASR_SERVICE.ORDER_ASR]: "orderAsr",
  [ASR_SERVICE.SOAP_ASR]: "soapAsr",
};

export const CONSENT_TYPE = {
  TERMS_CONDITIONS: "terms-conditions",
  PRIVACY_POLICIES: "privacy-policies",
};

export const SUBMIT_CONSENT_TYPE = {
  [CONSENT_TYPE.TERMS_CONDITIONS]: "terms",
  [CONSENT_TYPE.PRIVACY_POLICIES]: "privacy_policies",
};
