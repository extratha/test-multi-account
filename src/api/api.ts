import axios from "axios";

import { ENV } from "@/constant";
import {
  ChangePasswordRequest,
  ConsentResult,
  ConsentResultLatest,
  ExampleDataResult,
  InterpretResult,
  SubmitLabInterpretsRequest,
  SubmitLabInterpretsResult,
} from "@/types/model.api";
import { appendHeaders, onRefreshToken } from "./interceptors";

export const apiAxios = axios.create({ baseURL: ENV.BASE_API_URL });

apiAxios.interceptors.request.use(appendHeaders);
apiAxios.interceptors.response.use((response) => response, onRefreshToken);

export const submitChangePassword = (data: ChangePasswordRequest) => {
  return apiAxios.post("/auth/change-password", data);
};

export const getLabExampleList = () => {
  return apiAxios.get<ExampleDataResult[]>("/lab/examples");
};

export const getLabExampleId = (id: string) => {
  return apiAxios.get<InterpretResult>(`/lab/examples/${id}`);
};

export const submitLabInterprets = (data: SubmitLabInterpretsRequest) => {
  return apiAxios.post<SubmitLabInterpretsResult>("/lab/interprets", data);
};

export const getLabInterpretsByTransactionId = (transactionId: string) => {
  return apiAxios.get<InterpretResult>(`/lab/interprets/${transactionId}`);
};

export const getConsent = (type: string) => {
  return apiAxios.get<ConsentResult>(`/consents/${type}`);
};

export const getConsentLatest = (type: string) => {
  return apiAxios.get<ConsentResultLatest>(`/consents/${type}/latest`);
};

export const submitConsent = (type: string, version: string) => {
  return apiAxios.post<ConsentResult>("/consents", { type, version });
};
