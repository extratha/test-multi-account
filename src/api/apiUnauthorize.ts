import axios from "axios";

import { ENV } from "@/constant";
import { ForgotPasswordRequest, LoginRequest, LoginResult, SetNewPasswordRequest } from "@/types/model.api";

export const apiUnauthorizedAxios = axios.create({ baseURL: ENV.BASE_API_URL });

export const submitLogin = (data: LoginRequest) => {
  return apiUnauthorizedAxios.post<LoginResult>("/auth/login", data);
};

export const submitForgotPassword = (data: ForgotPasswordRequest) => {
  return apiUnauthorizedAxios.post("/auth/forgot-password", data);
};

export const submitSetNewPassword = (data: SetNewPasswordRequest) => {
  return apiUnauthorizedAxios.post("/auth/reset-password", data);
};

export const getValidateResetPasswordToken = (token: string) => {
  return apiUnauthorizedAxios.get(`/auth/reset-password/validate-token?token=${token}`);
};

export const getPublicConsent = (consentType: string) => {
  return apiUnauthorizedAxios.get(`/consents/${consentType}/public`);
};
