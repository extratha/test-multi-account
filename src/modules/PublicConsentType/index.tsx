"use client";

import { Divider, styled, Typography } from "@mui/material";
import { useParams } from "next/navigation";
import { useEffect, useMemo, useState } from "react";

import { getPublicConsent } from "@/api/apiUnauthorize";
import ConsentContent from "@/components/ConsentContent";
import FullScreenLoading from "@/components/Loading/FullScreenLoading";
import { Page } from "@/components/Page";
import ConsentPage from "@/components/Page/ConsentPage";
import PageNotFound from "@/components/Page/PageNotFound";
import { CONSENT_TYPE } from "@/constant";
import useTranslation from "@/locales/useLocale";
import { ConsentResult } from "@/types/model.api";

export interface PublicConsentTypeParams {
  [key: string]: string | string[];
  consentType: string;
}

const TitleDivider = styled(Divider)({
  margin: "8px 0px 24px",
});

const PublicConsentType = () => {
  const { consentType } = useParams<PublicConsentTypeParams>();
  const { translation } = useTranslation();

  const [consent, setConsent] = useState<ConsentResult>();
  const [isLoading, setIsLoading] = useState(true);
  const [isNotFound, setIsNotFound] = useState(false);

  const title = useMemo(() => {
    switch (consentType) {
      case CONSENT_TYPE.TERMS_CONDITIONS:
        return translation("Common.pages.termsAndConditions");
      case CONSENT_TYPE.PRIVACY_POLICIES:
        return translation("Common.pages.privacyPolicy");
      default:
        return "";
    }
  }, [consentType]);

  const fetchConsent = async () => {
    try {
      setIsLoading(true);
      const response = await getPublicConsent(consentType);

      setConsent(response.data);
      setIsLoading(false);
    } catch (error) {
      const statusCode = error?.response?.status;
      setIsNotFound(statusCode === 404);
      setIsLoading(false);
    }
  };

  useEffect(() => {
    fetchConsent();
  }, []);

  return (
    <Page title={title}>
      {isLoading && <FullScreenLoading />}
      {isNotFound && <PageNotFound />}
      {consent && (
        <ConsentPage>
          <Typography variant="titleLargeBold" data-testid="public-consent-type-title">
            {title}
          </Typography>
          <TitleDivider />
          <ConsentContent name="public-consent-type" data={consent.consent} />
        </ConsentPage>
      )}
    </Page>
  );
};

export default PublicConsentType;
