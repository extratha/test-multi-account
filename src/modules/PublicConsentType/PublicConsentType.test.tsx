import { screen } from "@testing-library/react";
import MockAdapter from "axios-mock-adapter";

import { mockPublicPrivacyPolicy, mockPublicTermsAndCons } from "@/__mocks__/data";
import * as ApiUnauthorizedAxios from "@/api/apiUnauthorize";
import { apiUnauthorizedAxios } from "@/api/apiUnauthorize";
import { CONSENT_TYPE } from "@/constant";
import { spyUseParams } from "@/testUtils/navigation";
import { API, flushPromise, render } from "@/testUtils/testUtils";
import PublicConsentType, { PublicConsentTypeParams } from ".";

describe("PublicConsentType", () => {
  let params: PublicConsentTypeParams;
  let mockApiAdapter: MockAdapter;

  beforeEach(() => {
    params = { consentType: "terms-conditions" };
    spyUseParams(params);

    jest.spyOn(ApiUnauthorizedAxios, "getPublicConsent");

    mockApiAdapter = new MockAdapter(apiUnauthorizedAxios);
    mockApiAdapter.onGet(API.PUBLIC_TERMS_CONDITIONS).reply(200, mockPublicTermsAndCons);
    mockApiAdapter.onGet(API.PUBLIC_PRIVACY_POLICIES).reply(200, mockPublicPrivacyPolicy);
  });

  const renderPublicConsentType = async () => {
    const view = render(<PublicConsentType />);
    await flushPromise();
    return view;
  };

  it("should render correctly", async () => {
    const { asFragment } = await renderPublicConsentType();
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render terms and condition", async () => {
    params.consentType = "terms-conditions";

    await renderPublicConsentType();

    expect(ApiUnauthorizedAxios.getPublicConsent).toHaveBeenCalledWith(CONSENT_TYPE.TERMS_CONDITIONS);
    expect(screen.getByTestId("public-consent-type-title")).toHaveTextContent("ข้อกำหนดและเงื่อนไข");
  });

  it("should render privacy policy", async () => {
    params.consentType = "privacy-policies";

    await renderPublicConsentType();

    expect(ApiUnauthorizedAxios.getPublicConsent).toHaveBeenCalledWith(CONSENT_TYPE.PRIVACY_POLICIES);
    expect(screen.getByTestId("public-consent-type-title")).toHaveTextContent("นโยบายความเป็นส่วนตัว");
  });

  it("should redirect to not-found page when api consents error", async () => {
    mockApiAdapter.onGet(API.PUBLIC_TERMS_CONDITIONS).reply(404);
    await renderPublicConsentType();

    expect(screen.getByTestId("page-not-found")).toBeInTheDocument();
  });
});
