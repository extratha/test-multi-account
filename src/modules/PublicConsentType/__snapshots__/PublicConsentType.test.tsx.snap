// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicConsentType should render correctly 1`] = `
<DocumentFragment>
  <title>
    ข้อกำหนดและเงื่อนไข
  </title>
  .emotion-0 {
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: -1;
  background-color: #eceff1;
}

<div
    class="MuiStack-root emotion-0"
  />
  .emotion-0 {
  width: 100%;
  margin-left: auto;
  box-sizing: border-box;
  margin-right: auto;
  display: block;
  padding-left: 16px;
  padding-right: 16px;
  max-width: 1024px;
  margin: 0px auto;
  padding: 24px 16px;
}

@media (min-width:600px) {
  .emotion-0 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    max-width: 1200px;
  }
}

.emotion-1 {
  background-color: #fff;
  color: unset;
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 16px;
  box-shadow: none;
  padding: 36px 40px;
}

.emotion-2 {
  margin: 0;
  font-size: 20px;
  font-weight: 700;
  line-height: 28px;
}

.emotion-3 {
  margin: 8px 0px 24px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-width: 0;
  border-style: solid;
  border-color: rgba(0, 0, 0, 0.12);
  border-bottom-width: thin;
}

.emotion-4 {
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-4>:not(style):not(style) {
  margin: 0;
}

.emotion-4>:not(style)~:not(style) {
  margin-top: 24px;
}

.emotion-5 {
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 24px;
  border-radius: 12px;
  background-color: #FAFAFA;
}

.emotion-6 {
  font-size: 16px;
  line-height: 28px;
}

.emotion-6>h3:first-of-type,
.emotion-6>p:first-of-type {
  margin-top: 0px;
}

.emotion-6>h3:last-of-type,
.emotion-6>p:last-of-type {
  margin-bottom: 0px;
}

.emotion-6 ul {
  padding-left: 24px;
  list-style-type: disc;
}

.emotion-6 ol {
  list-style-type: none;
  counter-reset: item;
  padding: 0px;
}

.emotion-6 ol>li {
  display: table-row;
  counter-increment: item;
  border-collapse: separate;
  border-spacing: 0px 6px;
}

.emotion-6 ol>li:before {
  display: table-cell;
  content: counters(item, ".") ". ";
  padding-right: 6px;
  white-space: nowrap;
}

.emotion-6 ol>ol,
.emotion-6 ol>ul {
  padding-left: 24px;
}

.emotion-6 ol>ol>li>ol,
.emotion-6 ol>ul>li>ol,
.emotion-6 ol>ol>li>ul,
.emotion-6 ol>ul>li>ul {
  padding: 0px;
}

.emotion-6 li {
  margin: 16px 0px;
}

.emotion-6 li ol>li {
  border-collapse: separate;
  border-spacing: 0px 6px;
}

.emotion-6 li ol>li:before {
  content: counters(item, ".") " ";
  padding-right: 6px;
}

.emotion-6 table {
  width: 100%;
  border-collapse: collapse;
  table-layout: fixed;
  margin: 16px 0px;
}

.emotion-6 table li {
  margin: 0px;
}

.emotion-6 th {
  height: 48px;
  text-align: center;
  color: #ffffff;
  border: 1px solid #000000;
  border-right-color: #ffffff;
  background-color: #002060;
}

.emotion-6 th:last-child {
  border-color: #000000;
}

.emotion-6 td {
  border: 1px solid #000000;
  padding: 8px;
  vertical-align: top;
}

.emotion-8 {
  margin: 0;
  font-size: 18px;
  font-weight: 700;
  line-height: 24px;
  margin-bottom: 24px;
}

.emotion-9 {
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 1px solid #cfd8dc;
  background-color: #fff;
}

.emotion-9:first-of-type {
  border-radius: 12px 12px 0 0;
}

.emotion-9:last-of-type {
  border-radius: 0 0 12px 12px;
}

.emotion-10 {
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  cursor: pointer;
  padding: 24px;
}

.emotion-12 {
  max-width: 52px;
  max-height: 52px;
}

.emotion-13 {
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-left: 12px;
}

.emotion-13>:not(style):not(style) {
  margin: 0;
}

.emotion-13>:not(style)~:not(style) {
  margin-top: 4px;
}

.emotion-14 {
  margin: 0;
  font-size: 16px;
  font-weight: 500;
  line-height: 1.5;
}

.emotion-15 {
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-16 {
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 50%;
  padding: 8px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: rgba(0, 0, 0, 0.54);
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  overflow: visible;
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-16::-moz-focus-inner {
  border-style: none;
}

.emotion-16.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-16 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-16:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-16:hover {
    background-color: transparent;
  }
}

.emotion-16.Mui-disabled {
  background-color: transparent;
  color: rgba(0, 0, 0, 0.26);
}

.emotion-17 {
  width: 24px;
  height: 24px;
}

.emotion-18 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-19 {
  height: 0;
  overflow: hidden;
  -webkit-transition: height 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: height 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  visibility: hidden;
}

.emotion-20 {
  display: flex;
  width: 100%;
}

.emotion-21 {
  width: 100%;
}

.emotion-22 {
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 16px;
}

<div
    class="MuiContainer-root MuiContainer-maxWidthLg emotion-0"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 emotion-1"
    >
      <span
        class="MuiTypography-root MuiTypography-titleLargeBold emotion-2"
        data-testid="public-consent-type-title"
      >
        ข้อกำหนดและเงื่อนไข
      </span>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth emotion-3"
      />
      <div
        class="MuiStack-root emotion-4"
      >
        <div
          class="MuiStack-root emotion-5"
        >
          <div
            class="emotion-6"
            data-testid="public-consent-type-content"
          >
            <h2>
              Terms and condition
            </h2>
          </div>
        </div>
        <div
          class="MuiStack-root emotion-5"
          data-testid="public-consent-type-service"
        >
          <span
            class="MuiTypography-root MuiTypography-titleBold emotion-8"
          >
            All Services
          </span>
          <div
            class="MuiStack-root emotion-9"
          >
            <div
              class="MuiStack-root emotion-10"
              data-testid="public-consent-type-service-0"
            >
              <div
                class="MuiBox-root emotion-11"
              >
                <img
                  alt=""
                  class="emotion-12"
                  data-testid="public-consent-type-service-0-logo"
                  src="/service/logo/0"
                />
              </div>
              <div
                class="MuiStack-root emotion-13"
              >
                <span
                  class="MuiTypography-root MuiTypography-bodyMedium emotion-14"
                  data-testid="public-consent-type-service-0-title"
                >
                  Service Title 0
                </span>
                <svg
                  data-testid="public-consent-type-service-0-detail-icon"
                />
              </div>
              <div
                class="MuiStack-root emotion-15"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-16"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    class="emotion-17"
                  />
                  <span
                    class="MuiTouchRipple-root emotion-18"
                  />
                </button>
              </div>
            </div>
            <div
              class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden emotion-19"
              style="min-height: 0px;"
            >
              <div
                class="MuiCollapse-wrapper MuiCollapse-vertical emotion-20"
              >
                <div
                  class="MuiCollapse-wrapperInner MuiCollapse-vertical emotion-21"
                >
                  <div
                    class="MuiStack-root emotion-22"
                    data-testid="public-consent-type-service-0-content"
                  >
                    <div
                      class="emotion-6"
                      data-testid="public-consent-type-content"
                    >
                      Consent Content 0
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
